plugins/guests/darwin/cap/change_host_name.rb

```diff
@@ -6,7 +6,7 @@ def self.change_host_name(machine, name)
           if !machine.communicate.test("hostname -f | grep '^#{name}$' || hostname -s | grep '^#{name}$'")
             machine.communicate.sudo("scutil --set ComputerName #{name}")
             machine.communicate.sudo("scutil --set HostName #{name}")
-            machine.communicate.sudo("scutil --set LocalHostName #{name}")
+            machine.communicate.sudo("scutil --set LocalHostName #{name.split('.')[0]}")
```

Comment:
Hrm. Let's just gsub the dots out then, since that's what OSX is doing. Maybe leave a comment in the code explaining that though.